### ISSUE TITLE
pylint ignore type annotation for property-like attributes

### ISSUE THREAD

**Issue by programminglaboratorys** [microsoft/vscode-pylint#602]
## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): Python 3.10.11
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): venv
- Operating system (and version): Windows 10 Pro (10.0.19045)
- Version of tool extension you are using: ms-python.pylint 2025.2.0 pre-release

## Behaviour
raises no-member when clearly the type is Font
### Expected Behavior
access get_height without any pylint warnings

### Actual Behavior
clean execution with no errors

## Reproduction Steps:
1- create a class with property-like attributes 
2- type annotation that attribute
done, it would think that attribute is a lambda no matter what it's type annotation according to it's warning

here a code that would raise the exact error
```python
""" example of pylint confuseion """
from dataclasses import dataclass

# pylint: disable=too-few-public-methods
def instantiate(*args, **kwargs):
    """ A decorator that instantiates the class with the given arguments. """
    def decorator(cls):
        return cls(*args, **kwargs)
    return decorator

@dataclass
class Font:
    """ dummy font class """
    path: str
    size: int
    def get_height(self):
        """ get font height """
        return self.size

class LazyAttributes:
    """
    A class that loads attributes lazily when they are accessed
    """
    _loaded_attributes = {}
    def __getattribute__(self, name):
        if name.startswith("_"):
            return super().__getattribute__(name)
        attr = super().__getattribute__(name)
        if callable(attr):
            print("attribute", name, "value", attr, flush=True)
            if name not in self._loaded_attributes:
                self._loaded_attributes[name] = attr()
            return self._loaded_attributes[name]
        return attr

@instantiate()
class FontBank(LazyAttributes):
    """ A class that loads fonts lazily when they are accessed """
    arialnb_font: Font = lambda _: Font("textures/fonts/ARIALNB.TTF", 20)
    lcd_font: Font = lambda _: Font("textures/fonts/LcdSolid.ttf", 30)

# Function '<lambda>' has no 'get_height' memberPylintE1101:no-member
print(FontBank.arialnb_font.get_height())
```

## Logs:

<details>
  <summary>Click here for detailed logs</summary>
<code>
  2025-03-13 15:25:12.924 [info] Name: Pylint
2025-03-13 15:25:12.924 [info] Module: pylint
2025-03-13 15:25:12.926 [debug] Configuration: {"name":"Pylint","module":"pylint"}
2025-03-13 15:25:12.926 [info] Python extension loading
2025-03-13 15:25:12.926 [info] Waiting for interpreter from Python extension.
2025-03-13 15:25:13.872 [info] No interpreter found from setting pylint.interpreter
2025-03-13 15:25:13.872 [info] Getting interpreter from ms-python.python extension for workspace c:\Users\ACER\AppData\Local\Programs\Microsoft VS Code Insiders
2025-03-13 15:25:13.879 [info] Python extension loaded
2025-03-13 15:25:13.903 [info] Interpreter from ms-python.python extension for c:\Users\ACER\AppData\Local\Programs\Microsoft VS Code Insiders: c:\Program Files\Python\Python310\python.exe
2025-03-13 15:25:14.070 [info] Server run command: c:\Program Files\Python\Python310\python.exe c:\Users\ACER\.vscode-insiders\extensions\ms-python.pylint-2025.2.0\bundled\tool\lsp_server.py
2025-03-13 15:25:14.073 [info] Server: Start requested.
2025-03-13 15:25:14.073 [debug] Server State: Starting
2025-03-13 15:25:18.036 [info] CWD Server: c:\Users\ACER\AppData\Local\Programs\Microsoft VS Code Insiders
2025-03-13 15:25:18.055 [info] Settings used to run Server:
[]

2025-03-13 15:25:18.062 [info] Global settings:
{
    "cwd": "${workspaceFolder}",
    "enabled": true,
    "workspace": "C:\\Users\\ACER\\AppData\\Local\\Programs\\Microsoft VS Code Insiders",
    "args": [],
    "severity": {
        "convention": "Information",
        "error": "Error",
        "fatal": "Error",
        "refactor": "Hint",
        "warning": "Warning",
        "info": "Information"
    },
    "path": [],
    "ignorePatterns": [],
    "interpreter": [],
    "importStrategy": "useBundled",
    "showNotifications": "off",
    "extraPaths": []
}

2025-03-13 15:25:18.097 [info] sys.path used to run Server:
   c:\Users\ACER\AppData\Local\Programs\Microsoft VS Code Insiders
   c:\Users\ACER\.vscode-insiders\extensions\ms-python.pylint-2025.2.0\bundled\libs
   c:\Users\ACER\.vscode-insiders\extensions\ms-python.pylint-2025.2.0\bundled\tool
   c:\Program Files\Python\Python310\python310.zip
   c:\Program Files\Python\Python310\DLLs
   c:\Program Files\Python\Python310\lib
   c:\Program Files\Python\Python310
   C:\Users\ACER\AppData\Roaming\Python\Python310\site-packages
   c:\Program Files\Python\Python310\lib\site-packages
   c:\Program Files\Python\Python310\lib\site-packages\tiblocks-0.1.0-py3.10.egg
   c:\Program Files\Python\Python310\lib\site-packages\win32
   c:\Program Files\Python\Python310\lib\site-packages\win32\lib
   c:\Program Files\Python\Python310\lib\site-packages\Pythonwin
2025-03-13 15:25:18.203 [info] c:\Program Files\Python\Python310\python.exe -m pylint --version
2025-03-13 15:25:18.300 [info] CWD Linter: C:\Users\ACER\AppData\Local\Programs\Microsoft VS Code Insiders
2025-03-13 15:25:19.043 [info] 
pylint 3.3.4
astroid 3.3.8
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]

2025-03-13 15:25:19.045 [info] Version info for linter running for C:\Users\ACER\AppData\Local\Programs\Microsoft VS Code Insiders:
pylint 3.3.4
astroid 3.3.8
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]

2025-03-13 15:25:19.046 [info] SUPPORTED pylint>=2.12.2
FOUND pylint==3.3.4

2025-03-13 15:25:19.103 [debug] Server State: Running
2025-03-13 15:25:19.147 [info] [Trace - 3:25:19 PM] Sending request 'textDocument/codeAction - (1)'.
2025-03-13 15:25:19.147 [info] Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/ACER/Desktop/myproject/fnaf-world-save-editor/__pycache__/example-pylint-confuseion.py"
    },
    "range": {
        "start": {
            "line": 0,
            "character": 0
        },
        "end": {
            "line": 0,
            "character": 0
        }
    },
    "context": {
        "diagnostics": [],
        "triggerKind": 2
    }
}

2025-03-13 15:25:19.167 [info] [Trace - 3:25:19 PM] Received notification 'window/logMessage'.
2025-03-13 15:25:19.167 [info] Params: {
    "type": 4,
    "message": "c:\\Program Files\\Python\\Python310\\python.exe -m pylint --reports=n --output-format=json --from-stdin C:\\Users\\ACER\\Desktop\\myproject\\fnaf-world-save-editor\\__pycache__\\example-pylint-confuseion.py"
}

2025-03-13 15:25:19.167 [info] c:\Program Files\Python\Python310\python.exe -m pylint --reports=n --output-format=json --from-stdin C:\Users\ACER\Desktop\myproject\fnaf-world-save-editor\__pycache__\example-pylint-confuseion.py
2025-03-13 15:25:19.176 [info] [Trace - 3:25:19 PM] Received notification 'window/logMessage'.
2025-03-13 15:25:19.176 [info] Params: {
    "type": 4,
    "message": "CWD Linter: C:\\Users\\ACER\\Desktop\\myproject\\fnaf-world-save-editor\\__pycache__"
}

2025-03-13 15:25:19.177 [info] CWD Linter: C:\Users\ACER\Desktop\myproject\fnaf-world-save-editor\__pycache__
2025-03-13 15:25:21.035 [info] [Trace - 3:25:21 PM] Received notification 'window/logMessage'.
2025-03-13 15:25:21.036 [info] Params: {
    "type": 4,
    "message": "file:///c%3A/Users/ACER/Desktop/myproject/fnaf-world-save-editor/__pycache__/example-pylint-confuseion.py :\r\n[\n    {\n        \"type\": \"convention\",\n        \"module\": \"example-pylint-confuseion\",\n        \"obj\": \"\",\n        \"line\": 1,\n        \"column\": 0,\n        \"endLine\": null,\n        \"endColumn\": null,\n        \"path\": \"example-pylint-confuseion.py\",\n        \"symbol\": \"invalid-name\",\n        \"message\": \"Module name \\\"example-pylint-confuseion\\\" doesn't conform to snake_case naming style\",\n        \"message-id\": \"C0103\"\n    },\n    {\n        \"type\": \"error\",\n        \"module\": \"example-pylint-confuseion\",\n        \"obj\": \"\",\n        \"line\": 43,\n        \"column\": 6,\n        \"endLine\": 43,\n        \"endColumn\": 38,\n        \"path\": \"example-pylint-confuseion.py\",\n        \"symbol\": \"no-member\",\n        \"message\": \"Function '<lambda>' has no 'get_height' member\",\n        \"message-id\": \"E1101\"\n    }\n]\n"
}

2025-03-13 15:25:21.036 [info] file:///c%3A/Users/ACER/Desktop/myproject/fnaf-world-save-editor/__pycache__/example-pylint-confuseion.py :
[
    {
        "type": "convention",
        "module": "example-pylint-confuseion",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "example-pylint-confuseion.py",
        "symbol": "invalid-name",
        "message": "Module name \"example-pylint-confuseion\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "error",
        "module": "example-pylint-confuseion",
        "obj": "",
        "line": 43,
        "column": 6,
        "endLine": 43,
        "endColumn": 38,
        "path": "example-pylint-confuseion.py",
        "symbol": "no-member",
        "message": "Function '<lambda>' has no 'get_height' member",
        "message-id": "E1101"
    }
]

2025-03-13 15:25:21.101 [info] [Trace - 3:25:21 PM] Received notification 'textDocument/publishDiagnostics'.
2025-03-13 15:25:21.101 [info] Params: {
    "uri": "file:///c%3A/Users/ACER/Desktop/myproject/fnaf-world-save-editor/__pycache__/example-pylint-confuseion.py",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 0,
                    "character": 0
                },
                "end": {
                    "line": 0,
                    "character": 0
                }
            },
            "message": "Module name \"example-pylint-confuseion\" doesn't conform to snake_case naming style",
            "severity": 3,
            "code": "C0103:invalid-name",
            "codeDescription": {
                "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/invalid-name.html"
            },
            "source": "Pylint"
        },
        {
            "range": {
                "start": {
                    "line": 42,
                    "character": 6
                },
                "end": {
                    "line": 42,
                    "character": 38
                }
            },
            "message": "Function '<lambda>' has no 'get_height' member",
            "severity": 1,
            "code": "E1101:no-member",
            "codeDescription": {
                "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/error/no-member.html"
            },
            "source": "Pylint"
        }
    ]
}

2025-03-13 15:25:21.126 [info] [Trace - 3:25:21 PM] Received response 'textDocument/codeAction - (1)' in 1979ms.
2025-03-13 15:25:21.126 [info] Result: []

2025-03-13 15:25:21.377 [info] [Trace - 3:25:21 PM] Sending request 'textDocument/codeAction - (2)'.
2025-03-13 15:25:21.378 [info] Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/ACER/Desktop/myproject/fnaf-world-save-editor/__pycache__/example-pylint-confuseion.py"
    },
    "range": {
        "start": {
            "line": 0,
            "character": 0
        },
        "end": {
            "line": 0,
            "character": 0
        }
    },
    "context": {
        "diagnostics": [
            {
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 0
                    }
                },
                "message": "Module name \"example-pylint-confuseion\" doesn't conform to snake_case naming style",
                "code": "C0103:invalid-name",
                "codeDescription": {
                    "href": "https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/invalid-name.html"
                },
                "severity": 3,
                "source": "Pylint"
            }
        ],
        "triggerKind": 2
    }
}

2025-03-13 15:25:21.428 [info] [Trace - 3:25:21 PM] Received response 'textDocument/codeAction - (2)' in 51ms.
2025-03-13 15:25:21.429 [info] Result: []
</code>
</details>

## Outcome When Attempting Debugging Steps:

Did running it from the command line work? No, "No module named pylint"

## Extra Details

- What other Python-related extensions are you using? I'm using: ms-python.vscode-pylance, ms-python.python

**Comment by karthiknadig** [comment]
This should be filed on `pylint` the library. This repo is for the VS Code extension that is a thin wrapper around that library to expose the features to VS Code. File it here https://github.com/pylint-dev/pylint/issues
